libs/category/src/effects/category.effects.ts

```diff
@@ -1,38 +1,73 @@
 import { Injectable, Inject } from '@angular/core';
 import { Actions, Effect, ofType } from '@ngrx/effects';
-import { switchMap, catchError } from 'rxjs/operators';
+import { switchMap, catchError, withLatestFrom } from 'rxjs/operators';
 import { of, Observable } from 'rxjs';
+import { Store, select } from '@ngrx/store';
 
 import { DaffProductGridLoadSuccess } from '@daffodil/product';
 
 import { 
   DaffCategoryActionTypes, 
   DaffCategoryLoad, 
   DaffCategoryLoadSuccess, 
-  DaffCategoryLoadFailure } from '../actions/category.actions';
+  DaffCategoryLoadFailure, 
+  DaffChangeCategoryPageSize} from '../actions/category.actions';
 import { DaffCategoryDriver } from '../drivers/injection-tokens/category-driver.token';
 import { DaffCategoryServiceInterface } from '../drivers/interfaces/category-service.interface';
 import { DaffGetCategoryResponse } from '../models/get-category-response';
+import { selectSelectedCategoryId, selectCategoryPageSize } from '../selectors/category.selector';
+import { DaffCategoryRequest } from '../models/category-request';
 
 @Injectable()
 export class DaffCategoryEffects {
 
   constructor(
     private actions$: Actions,
-    @Inject(DaffCategoryDriver) private driver: DaffCategoryServiceInterface){}
+    @Inject(DaffCategoryDriver) private driver: DaffCategoryServiceInterface,
+    private store: Store<any>
+  ){}
 
   @Effect()
   loadCategory$ : Observable<any> = this.actions$.pipe(
     ofType(DaffCategoryActionTypes.CategoryLoadAction),
-    switchMap((action: DaffCategoryLoad) =>
-      this.driver.get(action.payload)
-        .pipe(
-          switchMap((resp: DaffGetCategoryResponse) => [
-            new DaffProductGridLoadSuccess(resp.products),
-            new DaffCategoryLoadSuccess(resp)
-          ]),
-          catchError(error => of(new DaffCategoryLoadFailure('Failed to load the category')))
-        )
+    switchMap((action: DaffCategoryLoad) => this.processCategoryGetRequest(action.payload))
+  )
+
+  @Effect()
+  changeCategoryPageSize$ : Observable<any> = this.actions$.pipe(
+    ofType(DaffCategoryActionTypes.ChangeCategoryPageSizeAction),
+    withLatestFrom(this.store.pipe(select(selectSelectedCategoryId))),
+    switchMap(([action, categoryId]: [DaffChangeCategoryPageSize, string]) =>
+      this.processCategoryGetRequest({
```

Comment:
If I'm not mistaken, changing the page size will clear out all filters and sorts orders, and this also appears true for the other effects as well.

Is this intentional?

Reply:
Yes, I just didn't bother with filters or sort options because they don't work in magento yet